fs_mark/fs_mark.py

```diff
@@ -0,0 +1,197 @@
+    MSG_LINE = r"(?P<line>(\s+\d+){3}.+?)\r+\n"
+    NO_SPC = r"Insufficient free space.+?\r+\n"
+    END = r"Average Files/sec:.+?\r+\n"
+    ERR = r".+?(?P<err>EIO|ENOSPC|ENOMEM).+?"
+    NO_CONT_BLOCK = r"(?P<msg>Lack of contiguous memory block of size \d+ bytes.+?)\r+\n"
+
+    while True:
+        if start_time and end_time:
+            loop_time = 3 * (end_time - start_time)
+            start_time = None
+            end_time = None
+
+        try:
+            idx = dut.expect([
+                NAME,
+                MSG_LINE,
+                NO_SPC,
+                END,
+                ERR,
+                NO_CONT_BLOCK
+            ], timeout=loop_time)
+            parsed = dut.match.groupdict()
+
+        except pexpect.TIMEOUT:
+            test_status = Status.FAIL
+            subresult = TestSubResult(name=str(test_name), msg='Timeout, probably fs hang-up', status=Status.FAIL)
+            result.add_subresult_obj(subresult)
+            break
+
+        if idx == 0:
+            test_name = parsed["name"]
+            start_time = time.time()
+        elif idx == 1:
+            got_time = True
+            splited_line = parsed["line"].split()
+            f_use = splited_line[0]
+            count = splited_line[1]
+            size = splited_line[2]
+            files_sec = splited_line[3]
+            app_overhead = splited_line[4]
+
+            line_dict = {}
+            line_dict['creatMin'] = splited_line[5]
+            line_dict['creatAvg'] = splited_line[6]
+            line_dict['creatMax'] = splited_line[7]
+            line_dict['writeMin'] = splited_line[8]
+            line_dict['writeAvg'] = splited_line[9]
+            line_dict['writeMax'] = splited_line[10]
+            line_dict['closeMin'] = splited_line[17]
+            line_dict['closeAvg'] = splited_line[18]
+            line_dict['closeMax'] = splited_line[19]
+
+            # If files are kept, fs_mark will unlink them so unlink time can be caught
+            if not ('-k' in test_name or '-L' in test_name or '-F' in test_name):
+                line_dict['unlinkMin'] = splited_line[20]
+                line_dict['unlinkAvg'] = splited_line[21]
+                line_dict['unlinkMax'] = splited_line[22]
+
+            # Write timings to file if exists
+            try:
+                with open(os.path.expanduser(logs_path), 'a') as file:
+                    if first_loop is True:
+                        first_loop = False
+                        file.write(test_name + ' ' + str(datetime.now()) + '\n')
+                    else:
+                        file.write('\n')
+                    yaml.dump(line_dict, file)
+
+            except FileNotFoundError:
+                pass
+
+            subresult = TestSubResult(name=str(test_name), status=Status.OK)
+
+            for name, value in line_dict.items():
+                if not t_data.timing_dict[(target, name)][0] <= int(value) <= t_data.timing_dict[(target, name)][1]:
```

Comment:
Is it what you meant?

```suggestion
                if int(value) < t_data.timing_dict[(target, name)][0] or int(value) > t_data.timing_dict[(target, name)][1]:
```